codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -265,18 +271,21 @@ public Http2Stream resetSent() {
         }
 
         @Override
-        public final Object setProperty(Object key, Object value) {
-            return data.put(key, value);
+        public final <K, V> V setProperty(PropertyKey<K, V> key, V value) {
+            checkNotNull(key, "key");
+            return properties.add((DefaultPropertyKey<K, V>) key, value);
```

Comment:
> @Scottmitch 
keep in mind that if we support PropertyKey independent of a connection, we are 1) also independent of an EventLoop and 2) such a "static" key has to be able to be created at any time and 3) we have to support channel re-registration between EventLoops.

@buchgr - Good points.  I'm thinking of the compressor/decompressor/flow control type use case where I'm not sure that the property key would be impacted by these events (let me know if I'm missing something).  It may not work for all use cases, but if the use case requires more dynamic / stateful property key registration we have that too.